src/python/build_management/build_manager.py

```diff
@@ -711,6 +713,57 @@ def setup(self):
 
     self._setup_application_path(build_update=build_update)
     self._post_setup_success(update_revision=build_update)
+
+    return True
+
+
+class FuchsiaBuild(Build):
+  """Represents a Fuchsia build."""
+
+  def __init__(self, base_build_dir, revision, target_weights=None):
+    super(FuchsiaBuild, self).__init__(base_build_dir, revision)
+    self.base_build_dir = ''
+    self.build_url = ''
+    self.build_dir_name = ''
+    self._build_dir = ''
+    self.target_weights = target_weights
+    self.revision = 0
+
+  @property
+  def build_dir(self):
+    return self._build_dir
+
+  def setup(self):
+    logs.log('Retrieving build %d.' % self.revision)
+
+    # Bucket for QEMU resources.
+    fuchsia_resources_dir = fuchsia.device.initialize_resources_dir()
+    environment.set_value('FUCHSIA_RESOURCES_DIR', fuchsia_resources_dir)
+
+    logs.log('Retrieved build r%d.' % self.revision)
+    logs.log('Extracting fuzz targets.' + fuchsia_resources_dir)
+    logs.log("Extracting fuzz targets." + fuchsia_resources_dir)
+    environment.set_value('FUCHSIA_DIR',
+                          os.path.join(fuchsia_resources_dir, 'build'))
+
+    symbolize_path = os.path.join(fuchsia_resources_dir, 'build', 'zircon',
+                                  'prebuilt', 'downloads', 'symbolize')
+    os.chmod(symbolize_path, 0o777)
+    llvm_symbolizer_path = os.path.join(fuchsia_resources_dir, 'build',
+                                        'buildtools', 'linux-x64', 'clang',
+                                        'bin', 'llvm-symbolizer')
+    os.chmod(llvm_symbolizer_path, 0o777)
+
+    host = Host.from_dir(
+        os.path.join(fuchsia_resources_dir, 'build', 'out', 'default'))
+    fuzz_targets = Fuzzer.filter(host.fuzzers, '')
+    # TODO(flowerhack): Get list of fuzz targets and pass to the actual
+    # randomizer, instead of using random.choice().
+    fuzz_target = random.choice(fuzz_targets)
+
+    environment.set_value('FUZZ_TARGET', str(fuzz_target))
+    logs.log(
+        str('Extracted fuzz target ' + fuzz_target[0] + '/' + fuzz_target[1]))
```

Comment:
I'm a bit confused by this line. Is this just taking the first and second characters of `fuzz_target`?

Reply:
oh, whups.  it's retrieved as a tuple, but it makes more sense to coerce it to a string and *then* print things... fixed now